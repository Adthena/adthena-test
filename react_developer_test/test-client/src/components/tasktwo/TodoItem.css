.todo {
  text-decoration: line-through;
}
.todo-item {
  margin-top: 15px;
}
