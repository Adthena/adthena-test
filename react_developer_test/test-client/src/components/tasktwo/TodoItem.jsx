
import React from 'react';

const TodoItem = (props) => {
    return (
        <li className={!props.complete ? 'todo' : ''}>
            {props.title}
            {props.userId}
        </li>)
};

export default TodoItem;
