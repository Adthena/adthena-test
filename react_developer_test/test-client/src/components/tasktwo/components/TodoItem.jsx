
import React from 'react';
import PropTypes from 'prop-types'

const TodoItem = (props) => {
    return (
        <li className={!props.completed ? 'todo' : ''}>
            {props.title}
        </li>)
};

TodoItem.propTypes = {
    title: PropTypes.string.isRequired,
    completed: PropTypes.bool.isRequired
};

export default TodoItem;
