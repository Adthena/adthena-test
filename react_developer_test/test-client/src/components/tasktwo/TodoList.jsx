import React, { useState } from 'react';
import TodoItem from './TodoItem';
import './TodoList.css';

const TodoList = (props) => {
    return (
        <div className="todo-list">
            {props.todos && props.todos.map(item =>
                <TodoItem {...item} key={item.id} />
            )}
        </div>
    );



}

export default TodoList;
