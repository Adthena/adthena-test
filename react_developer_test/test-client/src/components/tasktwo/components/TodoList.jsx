import React, { useState } from 'react';
import TodoItem from './TodoItem';
import ClipLoader from "react-spinners/ClipLoader";

const TodoList = (props) => {
    return (
        <>
            <div className="todo-list">
                {props.todos && props.todos.map(item =>
                    <TodoItem {...item} key={item.id} />
                )}
            </div>
            <ClipLoader
                size={150}
                loading={props.loading}
            />
        </>
    );



}

export default TodoList;
