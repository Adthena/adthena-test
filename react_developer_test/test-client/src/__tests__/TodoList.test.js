import React from 'react';
import { shallow } from "enzyme";
import TodoList from '../components/tasktwo/components/TodoList';
import TodoItem from '../components/tasktwo/components/TodoItem';

it('renders correctly', () => {
    const todos = [{
        "userId": 1,
        "id": 1,
        "title": "delectus aut autem",
        "completed": false
    }, {
        "userId": 1,
        "id": 2,
        "title": "delectus aut autem",
        "completed": false
    }];
    const rendered = shallow(<TodoList todos={todos} />);
    expect(rendered.find(TodoItem)).toHaveLength(2);
    expect(rendered).toMatchSnapshot();
})
