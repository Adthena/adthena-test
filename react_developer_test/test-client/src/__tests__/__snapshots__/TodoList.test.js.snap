// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <div
    className="todo-list"
  >
    <TodoItem
      completed={false}
      id={1}
      key="1"
      title="delectus aut autem"
      userId={1}
    />
    <TodoItem
      completed={false}
      id={2}
      key="2"
      title="delectus aut autem"
      userId={1}
    />
  </div>
  <Loader
    color="#000000"
    css=""
    loading={true}
    size={150}
  />
</Fragment>
`;
